apis/r/tests/testthat/test-SOMADataFrame.R

```diff
@@ -54,14 +56,46 @@ test_that("SOMADataFrame creation", {
 
   # Subselecting columns
   expect_error(
-    sidf$read(column_names = "foo"),
-    "'column_names' must only contain non-index columns"
+    sidf$read(column_names = "foobar"),
+    "'column_names' must only contain valid dimension or attribute columns"
   )
 
   tbl1 <- sidf$read(column_names = "bar")
-  expect_true(tbl1$Equals(tbl0$SelectColumns(c(0L, 1L))))
+  expect_true(tbl1$Equals(tbl0$SelectColumns(1L)))
 
   # Attribute filters
   tbl1 <- sidf$read(value_filter = "bar < 5")
   expect_true(tbl1$Equals(tbl0$Filter(tbl0$bar < 5)))
 })
+
+test_that("SOMADataFrame read", {
+    tdir <- tempfile()
+    tgzfile <- system.file("raw-data", "soco-pbmc3k_processed-obs.tar.gz", package="tiledbsoma")
+    untar(tarfile = tgzfile, exdir = tdir)
+
+    uri <- file.path(tdir, "obs")
+
+    sdf <- SOMADataFrame$new(uri)
+    z <- sdf$read()
+    tbl <- tibble::as_tibble(z)
+    expect_equal(nrow(tbl), 2638L)
+    expect_equal(ncol(tbl), 6L)
+
+    columns <- c("n_counts", "n_genes", "louvain")
+    sdf <- SOMADataFrame$new(uri)
+    z <- sdf$read(column_names=columns)
+    tbl <- tibble::as_tibble(z)
+    expect_equal(ncol(tbl), 3L)
+    expect_equal(colnames(tbl), columns)
```

Comment:
```suggestion
    tbl <- sdf$read(column_names=columns)
    expect_equal(tbl$num_columns, 3L)
    expect_equal(tbl$ColumnNames(), columns)
```

Definitely a nit but I think we could simplify here by using `arrow`'s methods directly and avoid the need to convert w/ tibble.

Reply:
Sure but it's neither here nor there as it is in the test layer. I should probably test some content.

Reply:
Done too.